java/src/main/java/com/logicalclocks/hsfs/FeatureStore.java

```diff
@@ -76,7 +71,7 @@ public FeatureStore() {
    */
   public FeatureGroup getFeatureGroup(@NonNull String name, @NonNull Integer version)
```

Comment:
there are many methods, like this on, that should be move to the base class. Basically everything should be in the base class, except for things that use engine or spark directly (for those add abstract method in the base and here add `@Overwrite` annotation).

Reply:
I think we need to leave `FeatureGroup` in hsfs as its legacy type FG that will be only used for Spark. But I will create abstract methids for `SreamFeatureGroups`

Reply:
OK I moved them as well